communication/mavlink_waypoint_handler.cpp

```diff
@@ -517,1514 +385,285 @@ void Mavlink_waypoint_handler::clear_waypoint_list(Mavlink_waypoint_handler* way
-        waypoint_handler->waypoint_list[waypoint_handler->current_waypoint_index_].current = 1;
-        waypoint_handler->current_waypoint_ = waypoint_handler->waypoint_list[waypoint_handler->current_waypoint_index_];
-        waypoint_handler->waypoint_coordinates_ = waypoint_handler_set_waypoint_from_frame(&waypoint_handler->current_waypoint_,
-                                                                                            waypoint_handler->ins_.origin(),
-                                                                                            &waypoint_handler->navigation_.dubin_state);
-
-        mavlink_message_t msg;
-        mavlink_msg_mission_current_pack(waypoint_handler->mavlink_stream_.sysid(),
-                                         waypoint_handler->mavlink_stream_.compid(),
-                                         &msg,
-                                         waypoint_handler->current_waypoint_index_);
-        waypoint_handler->mavlink_stream_.send(&msg);
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_REQUEST_LIST; // 43
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &send_count;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
+    init_success &= message_handler_.add_msg_callback(&callback);
 
-        waypoint_handler->state_.nav_plan_active = true;
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_COUNT; // 44
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &receive_count;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
+    init_success &= message_handler_.add_msg_callback(&callback);
 
-        result = MAV_RESULT_ACCEPTED;
-    }
-    else
-    {
-        result = MAV_RESULT_TEMPORARILY_REJECTED;
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_CLEAR_ALL; // 45
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &clear_waypoint_list;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
 
-        print_util_dbg_print("Not ready to switch to next waypoint. Either no waypoint loaded or flying towards one\r\n");
-    }
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_ACK; // 47
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &receive_ack_msg;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
+    init_success &= message_handler_.add_msg_callback(&callback);
+
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/884f8a7c28d63e27cb08fc13e70f957fc181272d

- `receive_waypoint` --> `mission_item_callback`
- `send_waypoint` --> `mission_request_callback`
- `send_count` --> `request_list_callback` 
- `receive_count` --> `mission_count_callback`
- `clear_waypoint_list` --> `mission_clear_all_callback`
- `receive_ack_msg` --> `mission_ack_callback`